src/main/java/com/hivemq/cli/commands/cli/TestBrokerCommand.java

```diff
@@ -0,0 +1,370 @@
+                final QosTestResult qos2TestResult = tester.testQos(MqttQos.EXACTLY_ONCE, qosTries);
+                final int qos2Publishes = qos2TestResult.getReceivedPublishes();
+                final float qos2Time = qos2TestResult.getTimeToReceivePublishes() / 1_000_000F;
+                System.out.printf("Received %d/%d publishes in %.2fms\n", qos2Publishes, qosTries, qos2Time);
+
+                // Test retain
+                System.out.print("\t- Retain: ");
+                System.out.println(tester.testRetain());
+
+                // Test if wildcard subscriptions are allowed
+                System.out.print("\t- Wildcard subscriptions: ");
+                final WildcardSubscriptionsTestResult wildcardSubscriptionsTestResult = tester.testWildcardSubscriptions();
+                if (wildcardSubscriptionsTestResult.isSuccess()) {
+                    System.out.println("OK");
+                } else {
+                    System.out.println("NO");
+                    System.out.print("\t\t> '+' Wildcard: ");
+                    System.out.println(wildcardSubscriptionsTestResult.getPlusWildcardTest());
+                    System.out.print("\t\t> '#' Wildcard: ");
+                    System.out.println(wildcardSubscriptionsTestResult.getHashWildcardTest());
+                }
+
+                // Test max payload size
+                System.out.print("\t- Payload size: ");
+                final PayloadTestResults payloadTestResults = tester.testPayloadSize(MAX_PAYLOAD_TEST_SIZE);
+                final int payloadSize = payloadTestResults.getPayloadSize();
+                if (payloadSize == MAX_PAYLOAD_TEST_SIZE) {
+                    System.out.println(">= " + payloadSize + " bytes");
+                } else {
+                    System.out.println(payloadSize + " bytes");
+                }
+
+                // Test max client id length
+                System.out.print("\t- Max. client id length: ");
+                final ClientIdLengthTestResults clientIdLengthTestResults = tester.testClientIdLength();
+                final int maxClientIdLength = clientIdLengthTestResults.getMaxClientIdLength();
+                System.out.println(maxClientIdLength + " bytes");
+            }
+
+        }
+    }
+
+    public void testMqtt3Features() {
+        final Mqtt3FeatureTester client = new Mqtt3FeatureTester(
+                host,
+                port,
+                authenticationOptions.getUser(),
+                authenticationOptions.getPassword(),
+                sslConfig,
+                timeOut
+        );
+
+        boolean mqtt3Support = false;
+
+        // Test if MQTT3 is supported
+        System.out.print("MQTT 3: ");
+        final Mqtt3ConnAck connAck = client.testConnect();
+        if (connAck == null) {
+            System.out.println("NO");
+        } else if (connAck.getReturnCode() == Mqtt3ConnAckReturnCode.SUCCESS) {
+            mqtt3Support = true;
```

Comment:
Imho it would be better to `return;` in the "NO" branches and remove the `if (mqtt3Support)`